### PR TITLE
Refresh the FAQ

### LINKED ISSUE #1862: Docs: Frequently Asked Questions have become stale
**What happened**:
The https://gateway-api.sigs.k8s.io/faq/ page has grown stale and in some cases conflicts with other more recently updated portions of our documentation.

**What you expected to happen**:

The following questions should all likely have less detailed answers on this page and instead link out to more frequently updated sections:

* **Q: What are the differences between Ingress and Gateway API?**
  Should include a link to https://gateway-api.sigs.k8s.io/guides/migrating-from-ingress/
* **Q: How can I expose custom capabilities through Gateway API?**
  Should be shortened significantly and include a link to https://gateway-api.sigs.k8s.io/references/policy-attachment/
* **Q: How should I think about alpha API versions?**
  Should include a link to https://gateway-api.sigs.k8s.io/concepts/versioning/
* **Q: Which Kubernetes versions are supported?**
  Should remove specific versions and instead link to https://gateway-api.sigs.k8s.io/concepts/versioning/#supported-versions

### PR DESCRIPTION
**Pull request by spencerhance**
**What type of PR is this?**

/kind documentation

**What this PR does / why we need it**:
Following the feedback from @robscott on #1862, this PR consolidates some of the FAQ answers and adds links to relevant docs.

**Which issue(s) this PR fixes**:

Fixes #1862

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: site-src/faq.md [hunk 1]
```diff
@@ -16,35 +16,25 @@
...
 *   **Q: How can I expose custom capabilities through Gateway API?<br>**
-    A: There is a lot of diversity in the networking and proxying
-    ecosystem, and many products will have features that are not directly
-    supported in the API.  However, there are a few mechanisms available
+    A: There are a few mechanisms available
     for extending the API with implementation-specific capabilities:
 
-    * Decorate Gateway API objects with implementation-specific objects. A
+    * The [Policy Attachment](https://gateway-api.sigs.k8s.io/references/policy-attachment/)
+      API allows you to decorate Gateway API objects with implementation-specific CRDs. A
```
Comment (robscott) [inline site-src/faq.md:31]:
```suggestion
      model allows you to decorate Gateway API objects with implementation-specific CRDs. A
```

File: site-src/faq.md [hunk 2]
```diff
@@ -16,35 +16,25 @@
...
-      places, the fields of Gateway API resources have the type
-      "string". This allows an implementation to support custom values
-      for those fields in addition to any values specified in the API.
+    * Use implementation-specific values for string fields in Gateway API resources.
 
-    * Use implementation-specific annotations. For some kinds of
-      configuration, implementations may choose to support custom
-      annotations on Gateway API objects. This approach continues
-      a proud tradition of extending Ingress objects.
+    * Use implementation-specific annotations on Gateway API objects.
```
Comment (robscott) [inline site-src/faq.md:37]:
```suggestion
    * As a last resort, use implementation-specific annotations on Gateway API objects.
```

**Review by robscott — COMMENTED** [review]
Thanks @spencerhance! A couple nits but otherwise LGTM.

/ok-to-test

File: site-src/faq.md (hunk 1 above)
Comment (spencerhance) [reply site-src/faq.md]: Thanks!  Updated.

File: site-src/faq.md (hunk 2 above)
Comment (spencerhance) [reply site-src/faq.md]: Thanks!  Updated.

**Pull request merged**